src/ripple/basics/tagged_integer.h

```diff
@@ -157,6 +163,19 @@ class tagged_integer
         return tagged_integer (lhs.m_value - rhs);
     }
 
+    template <class OtherInt>
+    friend
+    typename std::enable_if <
+        std::is_integral <OtherInt>::value &&
+            sizeof (OtherInt) <= sizeof (Int),
```

Comment:
Good observation.  I agree that `a%b <= a`, so there shouldn't be a concern. 

Regarding the mix of signed and unsigned, I'll need to think a bit more on how much protection we want vs. the ease-of-use.